File: 0078-subsets/0078-subsets.java
class Solution {
    private void solve(int[] nums, List<Integer> temp, List<List<Integer>> output, int i)
    {
        if(i==nums.length)
        {
            output.add(new ArrayList<>(temp));
            return;
        }
        solve(nums,temp,output,i+1);
        temp.add(nums[i]);
        solve(nums,temp,output,i+1);  
        temp.remove(temp.size()-1);    
    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> output=new ArrayList<>();
        List<Integer> temp=new ArrayList<>();
        solve(nums,temp,output,0);
        return output;
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def solve(self, nums: List[int], temp: List[int], output: List[List[int]], i: int):
        if(i==len(nums)):
            output.append(temp.copy())
            return
        self.solve(nums,temp,output,i+1)
        temp.append(nums[i])
        self.solve(nums,temp,output,i+1)
        temp.pop()  

    def subsets(self, nums: List[int]) -> List[List[int]]:
        output=[]
        temp=[]
        self.solve(nums,temp,output,0)
        return output